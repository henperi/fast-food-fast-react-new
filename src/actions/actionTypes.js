export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SIGNUP_USER = 'SIGNUP_USER';
export const SET_AUTH_USER = 'SET_AUTH_USER';
export const REMOVE_AUTH_USER = 'REMOVE_AUTH_USER';
export const SET_ERRORS = 'SET_ERRORS';
export const GET_MENU = 'GET_MENU';
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const UPDATE_CART_ITEM_QUANTITY = 'UPDATE_CART_ITEM_QUANTITY';
export const TOGGLE_CART_SLIDER = 'TOGGLE_CART_SLIDER';
export const FETCH_CART_ITEMS = 'FETCH_CART_ITEMS';
export const SET_VISIBLE_CART = 'SET_VISIBLE_CART';
