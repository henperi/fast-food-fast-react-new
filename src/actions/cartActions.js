import {
  ADD_TO_CART, REMOVE_FROM_CART, TOGGLE_CART_SLIDER, SET_CURRENT_CART,
} from './actionTypes';

export const setCurrentCart = payload => ({
  type: SET_CURRENT_CART,
  payload,
});

export const addToCart = (id, name, price) => ({
  type: ADD_TO_CART,
  id,
  name,
  price,
});

export const removeFromCart = id => ({ type: REMOVE_FROM_CART, id });

export const updateCartQty = (id, updates) => ({
  type: 'UPDATE_CART_ITEM_QUANTITY',
  updates,
  id,
});

export const emptyCart = () => {
  localStorage.removeItem('foodItems');
  return { type: 'EMPTY_CART' };
};

export const toggleCartSlider = () => ({ type: TOGGLE_CART_SLIDER });
