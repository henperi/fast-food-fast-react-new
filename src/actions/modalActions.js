// import { OPEN_MODAL } from './actionTypes';

// export const openModal = ({
//   id, cat, title, price, description,
// } = {}) => ({
//   type: OPEN_MODAL,
//   modal: {
//     isOpen: true,
//     id,
//     cat,
//     title,
//     price,
//     description,
//   },
// });

// export const emptyCart = () => ({ type: 'EMPTY_CART' });
