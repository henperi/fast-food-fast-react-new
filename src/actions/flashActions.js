import { SHOW_FLASH, REMOVE_FLASH } from './actionTypes';

export const showFlash = (data = {}) => {
  const { title = '', message = '', flashType = 'success' } = data;
  return {
    type: SHOW_FLASH,
    payload: {
      title,
      message,
      flashType,
    },
  };
};

export const removeFlash = () => ({
  type: REMOVE_FLASH,
});
