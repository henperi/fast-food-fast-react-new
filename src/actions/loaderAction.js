import { TOGGLE_LOADER } from './actionTypes';

const toggleLoader = () => ({
  type: TOGGLE_LOADER,
});

export default toggleLoader;
