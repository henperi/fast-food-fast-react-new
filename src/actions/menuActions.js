import { SET_MENU, SET_ERRORS, SET_FEATURED_MENU } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';

export const setMenu = (payload = {}) => ({
  type: SET_MENU,
  payload,
});

export const setFeaturedMenu = (payload = {}) => {
  const featuredMenu = payload.slice(0, 3);

  return {
    type: SET_FEATURED_MENU,
    payload: featuredMenu,
  };
};

export const getMenu = () => (dispatch) => {
  axiosInstance
    .get('/menu')
    .then((response) => {
      const { data } = response;
      dispatch({
        type: SET_MENU,
        payload: { menu: [...data.menu] },
      });
      dispatch(setFeaturedMenu(data.menu));
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export const updateMenu = () => {};
