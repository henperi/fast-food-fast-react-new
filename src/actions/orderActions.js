import { SET_ERRORS } from './actionTypes';
import axiosInstance from '../utils/axiosInstance';
import { emptyCart } from './cartActions';

const placeOrder = (cartItems = {}, handleModal) => (dispatch) => {
  console.log(cartItems);
  const orderData = {
    foodItems: cartItems,
  };
  axiosInstance
    .post('/orders', orderData)
    .then((response) => {
      const { data } = response;
      console.log(data);
      handleModal();
      dispatch(emptyCart());
    })
    .catch((errors) => {
      const { response = {}, request } = errors;
      console.log(errors);
      console.log(response);
      console.log('=======');
      console.log(request);

      if (response.data) {
        return dispatch({
          type: SET_ERRORS,
          payload: [...response.data.errors],
        });
      }

      return dispatch({
        type: SET_ERRORS,
        payload: [{ msg: 'An unknown error occurred, Please check your network and try again' }],
      });
    });
};

export default placeOrder;
