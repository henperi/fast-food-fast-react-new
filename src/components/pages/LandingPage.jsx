import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Header from '../ui/Header';
import Footer from '../ui/Footer';
import { DefaultFood } from '../ui/FoodCard';
import Hero from '../ui/Hero';
import AboutCard from '../ui/AboutCard';
import ContactCard from '../ui/ContactCard';
import { getMenu } from '../../actions/menuActions';
import { addToCart, toggleCartSlider } from '../../actions/cartActions';

export class LandingPage extends Component {
  state = {};

  componentDidMount() {
    const { getMenu: dispatchGetMenu } = this.props;
    dispatchGetMenu();
  }

  componentWillReceiveProps(nextProps) {
    nextProps.menu && this.setState({ ...nextProps.menu });
    nextProps.cart && this.setState({ cart: nextProps.cart });
  }

  handleAddToCart = (id, name, price) => {
    const {
      addToCart: dispatchAddToCart,
      toggleCartSlider: dispatchToggleCartSlidder,
    } = this.props;

    dispatchAddToCart(id, name, price);
    dispatchToggleCartSlidder();
  };

  render() {
    const { featuredMenu } = this.props;
    return (
      <div>
        <Header />
        <main>
          <Hero />
          <div className="p-h-5 text-center">
            <p>
              To enjoy amazing discounts and have a fun filled experience, login to your account!
            </p>
            <div>
              <Link to="/signin" className="btn btn-green">
                Login Now
              </Link>
              <Link to="/signup" className="btn btn-blue">
                Signup Now
              </Link>
            </div>
          </div>

          <div className="containerX">
            <section>
              <h2 className="text-center">Todays Featured Foods</h2>
              <div className="container food-menu">
                {featuredMenu.map(singleFood => (
                  <DefaultFood
                    key={singleFood.food_id}
                    {...singleFood}
                    handleAddToCart={this.handleAddToCart}
                  />
                ))}
              </div>
            </section>

            <section>
              <div className="card text-center">
                <h2>Wanna See Our Menu And All Food Items?</h2>
                <p>
                  <a href="users/foods.html" className="btn btn-blue btn-rounded btn-bg">
                    View Menu
                  </a>
                </p>
              </div>
              <div className="text-center">
                <p>
                  <a href="signin.html" className="btn btn-green">
                    Login Now!
                  </a>
                  <a href="signup.html" className="btn btn-blue">
                    Signup Now!
                  </a>
                </p>
              </div>
            </section>

            <section className="flex-h text-center">
              <AboutCard />
              <ContactCard />
            </section>
          </div>
        </main>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  featuredMenu: state.featuredMenu,
  cart: state.cart,
});

export default connect(
  mapStateToProps,
  { getMenu, addToCart, toggleCartSlider },
)(LandingPage);
