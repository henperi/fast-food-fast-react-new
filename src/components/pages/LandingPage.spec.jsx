/* eslint-disable no-undef */
import React from 'react';
import sinon from 'sinon';
import { LandingPage } from './LandingPage';
import { getMenu } from '../../actions/menuActions';
import { addToCart, toggleCartSlider, slideCart } from '../../actions/cartActions';
// import LoadingSpinner from '../ui/LoadingSpinner';
// import { loginUser } from '../../actions/userActions';
// import SignupForm from './SignupForm';

describe('The MenuPage Component Test Suite', () => {
  const wrapper = shallow(
    <LandingPage
      featuredMenu={[]}
      addToCart={addToCart}
      toggleCartSlider={toggleCartSlider}
      getMenu={getMenu}
      slideCart={slideCart}
    />,
  );
  it('should render the <MenuPage /> Component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(LandingPage)).toBeDefined();
  });

  it('should call the handleAddToCart', () => {
    wrapper.instance().handleAddToCart({ id: 5, name: 'Rice', price: 500 });
    expect(sinon.stub(jest.fn(wrapper.instance().props.addToCart()))).not.toHaveBeenCalled();
  });
});
