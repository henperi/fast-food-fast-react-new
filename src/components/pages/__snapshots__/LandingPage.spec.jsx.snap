// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MenuPage Component Test Suite should render the <MenuPage /> Component 1`] = `
<div>
  <Connect(Header) />
  <main>
    <Hero />
    <div
      className="p-h-5 text-center"
    >
      <p>
        To enjoy amazing discounts and have a fun filled experience, login to your account!
      </p>
      <div>
        <Link
          className="btn btn-green"
          replace={false}
          to="/signin"
        >
          Login Now
        </Link>
        <Link
          className="btn btn-blue"
          replace={false}
          to="/signup"
        >
          Signup Now
        </Link>
      </div>
    </div>
    <div
      className="containerX"
    >
      <section>
        <h2
          className="text-center"
        >
          Todays Featured Foods
        </h2>
        <div
          className="container food-menu"
        />
      </section>
      <section>
        <div
          className="card text-center"
        >
          <h2>
            Wanna See Our Menu And All Food Items?
          </h2>
          <p>
            <a
              className="btn btn-blue btn-rounded btn-bg"
              href="users/foods.html"
            >
              View Menu
            </a>
          </p>
        </div>
        <div
          className="text-center"
        >
          <p>
            <a
              className="btn btn-green"
              href="signin.html"
            >
              Login Now!
            </a>
            <a
              className="btn btn-blue"
              href="signup.html"
            >
              Signup Now!
            </a>
          </p>
        </div>
      </section>
      <section
        className="flex-h text-center"
      >
        <AboutCard />
        <ContactCard />
      </section>
    </div>
  </main>
  <Footer />
</div>
`;
